activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -360,28 +372,16 @@ def references(*args, **options)
       end
       alias :belongs_to :references
 
-      def new_column_definition(name, type, options) # :nodoc:
+      def new_column_definition(name, type, **options) # :nodoc:
         type = aliased_types(type.to_s, type)
-        column = create_column_definition name, type
-
-        column.limit       = options[:limit]
-        column.precision   = options[:precision]
-        column.scale       = options[:scale]
-        column.default     = options[:default]
-        column.null        = options[:null]
-        column.first       = options[:first]
-        column.after       = options[:after]
-        column.auto_increment = options[:auto_increment]
-        column.primary_key = type == :primary_key || options[:primary_key]
-        column.collation   = options[:collation]
-        column.comment     = options[:comment]
-        column.as          = options[:as]
-        column
+        options[:primary_key] ||= type == :primary_key
+        options[:null] = false if options[:primary_key]
```

Comment:
Hey @kamipo @jeremy , I'm wondering why this got added here? Just noticed this got moved from the sqlite3 adapter into abstract. Not sure why a column could not be null if you specify the `primary_key` option.

Sorry if you get a duplicate notification, for some reason my previous comment got deleted.